cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/StructuredTable.java

```diff
@@ -29,17 +30,18 @@
  * The schema of the table is fixed, and has to be specified in the
  * {@link StructuredTableSpecification} during the table creation.
  */
-// TODO: CDAP-14676 - Add IOException for table operations if needed when implementing SQL tables.
-public interface StructuredTable {
+public interface StructuredTable extends AutoCloseable {
```

Comment:
Is is `Closeable` or `AutoCloseable`? The subtle difference is `Closeable.close` throws `IOException`, while `AutoCloseable` throws a more generic `Exception`.

An extract from the javadoc of `AutoCloseable`:

```
     * <p>While this interface method is declared to throw {@code
     * Exception}, implementers are <em>strongly</em> encouraged to
     * declare concrete implementations of the {@code close} method to
     * throw more specific exceptions, or to throw no exception at all
     * if the close operation cannot fail.
```

Reply:
I changed it to `Closeable` since the majority of the method in this interface throws an IOException instead of a more generic `Exception`. And we can implement the close() method to throw a more specific exception